src/python_testing/TC_AVSM_2_3.py

```diff
@@ -0,0 +1,170 @@
+            ),
+            TestStep(
+                3,
+                "TH sends the SnapshotStreamModify command with SnapshotStreamID set to aStreamID. If WMARK is supported, set WaterMarkEnabled to !aWmark`and if OSD is supported, set OSDEnabled to `!aOSD in the command.",
+                "DUT responds with a SUCCESS status code.",
+            ),
+            TestStep(
+                4,
+                "TH reads AllocatedSnapshotStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the following: If WMARK is supported, verify WaterMarkEnabled == !aWmark. If OSD is supported, verify OSDEnabled == !aOSD.",
+            ),
+        ]
+
+    async def _precondition_one_allocated_snp_stream(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+        commands = Clusters.CameraAvStreamManagement.Commands
+
+        # First verify that SNP is supported
+        aFeatureMap = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.FeatureMap)
+        logger.info(f"Rx'd FeatureMap: {aFeatureMap}")
+        snpSupport = (aFeatureMap & cluster.Bitmaps.Feature.kSnapshot) > 0
+        asserts.assert_true(snpSupport, "Snapshot Feature is not supported.")
+
+        # Check if snapshot stream has already been allocated
+        aAllocatedSnapshotStreams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedSnapshotStreams
+        )
+        logger.info(f"Rx'd AllocatedSnapshotStreams: {aAllocatedSnapshotStreams}")
+        if len(aAllocatedSnapshotStreams) > 0:
+            return
+
+        # Allocate one for the test steps based on SnapshotCapabilities
+        aSnapshotCapabilities = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.SnapshotCapabilities
+        )
+        logger.info(f"Rx'd SnapshotCapabilities: {aSnapshotCapabilities}")
+
+        asserts.assert_greater(len(aSnapshotCapabilities), 0, "SnapshotCapabilities list is empty")
+        try:
+            snpStreamAllocateCmd = commands.SnapshotStreamAllocate(
+                imageCodec=aSnapshotCapabilities[0].imageCodec,
+                maxFrameRate=aSnapshotCapabilities[0].maxFrameRate,
+                minResolution=aSnapshotCapabilities[0].resolution,
+                maxResolution=aSnapshotCapabilities[0].resolution,
+                quality=90,
+            )
+            await self.send_single_cmd(endpoint=endpoint, cmd=snpStreamAllocateCmd)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+            pass
+
+    @async_test_body
```

Comment:
```suggestion
    @run_if_endpoint_matches(has_feature(Clusters.CameraAvStreamManagement,
                                                 Clusters.CameraAvStreamManagement.Bitmaps.Feature.kSnapshot)
```

Reply:
Thank you for your comments. We fixed it with our new commit.